src/main/java/opendcs/util/sql/WrappedConnection.java

```diff
@@ -0,0 +1,394 @@
+    {
+        Objects.requireNonNull(realConnection, "WrappedConnection cannot wrap a null connection");
+        Objects.requireNonNull(onClose, "WrappedConnections requires a valid Consumer for the close operation");
+        this.realConnection = realConnection;
+        this.onClose = onClose;
+        this.pool = pool;
+        if(pool.isPresent())
+        {
+            openTrace = new ArrayList<>();
+            StackTraceElement ste[] = Thread.currentThread().getStackTrace();
+            // start at 3, after this constructor and the getThread/Stack trace call
+            for(int i = 3; i < ste.length; i++)
+            {
+                openTrace.add(ste[i]);
+            }
+        }
+        
+    }
+
+    @Override
+    public boolean isWrapperFor(Class<?> iface) throws SQLException {
+        return realConnection.isWrapperFor(iface);
+    }
+
+    @Override
+    public <T> T unwrap(Class<T> iface) throws SQLException {
+        return realConnection.unwrap(iface);
+    }
+
+    @Override
+    public void abort(Executor executor) throws SQLException {
+        realConnection.abort(executor);
+    }
+
+    @Override
+    public void clearWarnings() throws SQLException {
+        realConnection.clearWarnings();
+    }
+
+    @Override
+    public void close() throws SQLException {
+        if (pool.isPresent())
+        {
+            openTrace = new ArrayList<>();
+            StackTraceElement ste[] = Thread.currentThread().getStackTrace();
+            // start at 3, after this constructor and the getThread/Stack trace call
+            for(int i = 3; i < ste.length; i++)
+            {
+                openTrace.add(ste[i]);
+            }
+            // send notice
+            end = ZonedDateTime.now();
+        }
+
+
+        onClose.accept(realConnection);
+    }
+
+    @Override
+    public void commit() throws SQLException {
+        realConnection.commit();
+    }
+
+    @Override
+    public Array createArrayOf(String typeName, Object[] elements) throws SQLException {
+        return realConnection.createArrayOf(typeName, elements);
+    }
+
+    @Override
+    public Blob createBlob() throws SQLException {
+        return realConnection.createBlob();
+    }
+
+    @Override
+    public Clob createClob() throws SQLException {
+        return realConnection.createClob();
+    }
+
+    @Override
+    public NClob createNClob() throws SQLException {
+        return realConnection.createNClob();
+    }
+
+    @Override
+    public SQLXML createSQLXML() throws SQLException {
+        return realConnection.createSQLXML();
+    }
+
+    @Override
+    public Statement createStatement() throws SQLException {
+        logPlainCreate();
+        return realConnection.createStatement();
+    }
+
+    private void logPlainCreate() {
+        log.debug2("Dev Msg: Plain create statement from:");
+        StackTraceElement stk[] = Thread.getAllStackTraces().get(Thread.currentThread());
```

Comment:
There should definitely be a `isLoggable()` check before snagging the stack traces. Also I would think that `Thread.currentThread().getStackTrace()` is cheaper than getting every thread's stack.

Reply:
oh wow. I was doing currentThread in other places but totally missed that one.